apps/src/storage/dataBrowser/dataVisualizer/CrossTabChart.jsx

```diff
@@ -104,43 +88,105 @@ class CrossTabChart extends React.Component {
       });
     });
 
+    // Our goal is uniform column widths, for all columns except the first one.
+    // There are a few steps that lead to successful layout here.
+    // 1. The table width is 100% to maximize available space.
+    // 2. The first column uses `white-space: nowrap` so it won't shrink too
+    //    small for its contents.
+    // 3. We set the rest of the columns to have the same percentage-width,
+    //    adding up to 99% of the table width (which is more than available space)
+    //    so after setting the first column width, the remaining columns
+    //    shrink to fit evenly.
+    const columnWidth = 99 / (columns.length - 1) + '%';
+
     const min = Math.min(...numericValues);
     const max = Math.max(...numericValues);
 
     return (
-      <div id={CROSS_TAB_CHART_AREA}>
-        <h1 style={styles.title}>{this.props.chartTitle}</h1>
-        <table>
-          <tbody>
-            <tr>
-              {columns.map(column => (
-                <th key={column} style={styles.headerCell}>
-                  {column}
-                </th>
-              ))}
-            </tr>
-            {chartData.map((record, id) => (
-              <tr key={id}>
-                {columns.map(column => {
-                  const value = record[column];
-                  const color =
-                    column === this.props.selectedColumn1
-                      ? 'white'
-                      : this.getColorForValue(value, min, max);
-                  const cellStyle = {...styles.cell, backgroundColor: color};
-                  return (
-                    <td key={column} style={cellStyle}>
-                      {value}
-                    </td>
-                  );
-                })}
-              </tr>
+      <div id={CROSS_TAB_CHART_AREA} style={wrapperStyle}>
+        <h1 style={chartTitleStyle}>{this.props.chartTitle}</h1>
+        <table style={tableStyle}>
+          <tr>
```

Comment:
nit: missing <tbody>

Reply:
The [HTML 5 spec says `tbody` isn't required](https://html.spec.whatwg.org/multipage/tables.html#the-tr-element), and without a `thead` element this seemed simpler.  Is there a particular reason you wanted one included here?

Reply:
I'm seeing this warning
![image](https://user-images.githubusercontent.com/8787187/77589707-6dddc400-6ea9-11ea-9185-167969f19168.png)

And it looks like the drone build is also failing for the same reason

Reply:
Ooooh, fascinating - there's a React warning for this.  [Conversation here](https://github.com/facebook/react/issues/5652) - it's important to provide this for React's virtual DOM because browsers implicitly insert a `tbody` element for you, creating a difference between your virtual DOM and the real one.  Still, some good comments in there about how this warning is misleading.

Reply:
Putting it back...